neuro_san/service/agent_server.py

```diff
@@ -195,3 +195,10 @@ def serve(self):
 
         self.notify_started.set()
         self.server_lifetime.run()
+
+    def stop(self):
+        """
+        Stop the server.
+        """
+        # pylint: disable=protected-access
+        self.server_lifetime._stop_serving()
```

Comment:
Add API method to stop gRPC server.
Of course, using internal method is bad form, but we can leave changes to leaf-server-common
for later PR if necessary.